server/channels/app/expirynotify.go

```diff
@@ -24,10 +24,11 @@ func (a *App) NotifySessionsExpired() error {
 	// Get all mobile sessions that expired within the last hour.
 	sessions, err := a.ch.srv.Store().Session().GetSessionsExpired(OneHourMillis, true, true)
 	if err != nil {
+		a.CountNotificationReason(model.NotificationStatusError, model.NotificationTypePush, model.NotificationReasonFetchError)
```

Comment:
Should we add what type of push notification are we handling here? In this case is a session notification, which may or may not be as important as a post notification or a clear notification.

Reply:
Should we count those separately in the metrics, or is that something I should just add in the logs? I'm not sure what the difference is, but sounds like it would still contribute to overall "notification health" so I think it's probably fine just as-is metrics-wise, but maybe worth adding in the logs?

Reply:
My main fear is background notifications (clear) vs normal notifications (messages).

Background notifications are more often not delivered to the app on iOS because of reasons we cannot control. That may create some perception of "unhealthiness" that is not real. That is why I would have them separate.

Reply:
It is expected, but there is not much we can do.

The clear notifications are the ones that clear notifications on a channel when the channel is read in a different client. Therefore, in general they are way more common than normal notifications.

Since there is nothing we can do about them not being delivered, and being way more than message notifications, they may mask issues with the "important" notifications: the messages.

Regarding handling them different, I am not sure what is your idea.

Reply:
Yes, they go through the push proxy.

The idea is that you need some way to remove the notification from the phone when the message gets read. You don't want to get to your phone and see a ton of notifications you have already read.

So you have to send a message to the phone so it removes those notification. And how that message looks like? A background notification.

I think it is valuable to have them in the metrics, but make a difference between them. But 0/5
For sure, the important ones are the ones that make a notification appear in the device: message and session push notifications.

Reply:
I think this might be something we follow up on separately actually - I think it might be important to know how often those "clear" messages are outright failing and group them in with notification health for now. If we find that to be a a major contributor we might want to revisit how we track it. But I think we can handle it later.